sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -665,7 +666,8 @@ private[sql] class SQLConf extends Serializable with CatalystConf with Logging {
   def variableSubstituteDepth: Int = getConf(VARIABLE_SUBSTITUTE_DEPTH)
 
   def warehousePath: String = {
-    getConf(WAREHOUSE_PATH).replace("${system:user.dir}", System.getProperty("user.dir"))
+    getConf(WAREHOUSE_PATH).replace("${system:user.dir}",
+      new File(System.getProperty("user.dir")).toURI.toURL.toString).replace("//", "/")
```

Comment:
Very interesting! In the function `normalizePath` of `Path`, it replaces double forward slashes by a single forward slash. 
https://hadoop.apache.org/docs/r2.4.1/api/src-html/org/apache/hadoop/fs/Path.html

We still need to do it here, even if `Path` can normalize it for us.